### DISCUSSION TITLE
Can llama.cpp support batched prompts?

### DISCUSSION THREAD

**Discussion by chosen-ox** [ggml-org/llama.cpp#10299] [Q&A] [answered]
I am new to llama.cpp. I browse discussions and issues to find how to inference multi requests together. For example, if there is only one prompt. We have a 2d array. If there are several prompts together, the input will be a matrix. In my opinion, processing several prompts together is faster than process them separately. Llama have provide [batched requests](https://github.com/meta-llama/llama/blob/8fac8befd776bc03242fe7bc2236cdb41b6c609c/llama/generation.py#L132). I wonder if llama.cpp have similar feature?

By the way, n_batch and n_ubatch in llama.cpp may refers to the chunk size in a single prompt. Not the number of prompts processed together.

**Comment by ExtReMLapin** [comment, marked as answer]
yes, easiest way to use it is with llama-server where you create multiple slots

Edit : 

Long story short, you set the number of parallel requests and parallel http threads to the batch size.

total context size = batch_size * individual context (4096 or 8192 etc)

**Reply by chosen-ox** [reply]
Does example directory have a correspond file I can learn about？Thank you!

**Reply by afsara-ben** [reply]
also looking for the same

**Reply by ExtReMLapin** [reply]
are you guys even reading and searching by yourself a little ??? https://github.com/ggml-org/llama.cpp/tree/master/examples/server
